src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -189,17 +192,35 @@ protected override async Task InitializeAsync(EventContext telemetryContext, Can
             this.ValidateParameters();
 
             DependencyPath workloadPackage = await this.GetPlatformSpecificPackageAsync(this.PackageName, cancellationToken);
-
             this.HPLDirectory = workloadPackage.Path;
 
-            await this.ConfigurePerformanceLibrary(telemetryContext, cancellationToken);
-            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, this.makeFileName));
-            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "setup", this.makeFileName));
+            DependencyPath performanceLibrariesPackage = await this.packageManager.GetPackageAsync(this.PerformanceLibrariesPackageName, cancellationToken)
+                    .ConfigureAwait(false);
 
-            await this.ExecuteCommandAsync("bash", "-c \"source make_generic\"", this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup"), telemetryContext, cancellationToken, runElevated: true);
-            await this.ConfigureMakeFileAsync(telemetryContext, cancellationToken);
-            await this.ExecuteCommandAsync("ln", $"-s {this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup", this.makeFileName)} {this.makeFileName}", this.HPLDirectory, telemetryContext, cancellationToken);
+            await this.ConfigurePerformanceLibrary(telemetryContext, cancellationToken);
 
+            if (this.CpuArchitecture == Architecture.X64 && this.PerformanceLibrary == "INTEL")
+            {
+                // Use PlatformSpecifics.Combine for paths instead of string concatenation
+                if (this.PerformanceLibraryVersion == "2024.2.2.17")
+                {
+                    string intelMklPath = "/opt/intel/oneapi/mkl/2024.2/share/mkl/benchmarks/mp_linpack";
+                    await this.ExecuteCommandAsync("cp", $"-r {intelMklPath} {this.intelPerfLibrariesPath}", this.HPLDirectory, telemetryContext, cancellationToken);
+                }
+                else if (this.PerformanceLibraryVersion == "2025.1.0.803")
```

Comment:
add else with error message mentioning what's supported version, also add supported version to document